packages/fxa-auth-server/lib/routes/support.js

```diff
@@ -18,19 +18,14 @@ module.exports = (log, db, config, customs, zendeskClient) => {
   const SUBSCRIPTIONS_MANAGEMENT_SCOPE =
     'https://identity.mozilla.com/account/subscriptions';
 
-  async function handleAuth(auth, fetchEmail = false) {
+  async function handleAuth(auth) {
     const scope = ScopeSet.fromArray(auth.credentials.scope);
     if (!scope.contains(SUBSCRIPTIONS_MANAGEMENT_SCOPE)) {
       throw error.invalidScopes('Invalid authentication scope in token');
     }
     const { user: uid } = auth.credentials;
-    let email;
-    if (!fetchEmail) {
-      ({ email } = auth.credentials);
-    } else {
-      const account = await db.account(uid);
-      ({ email } = account.primaryEmail);
-    }
+    const account = await db.account(uid);
+    const { email } = account.primaryEmail;
```

Comment:
We should go the other direction: always rely on the token for the email.

Reply:
Maybe not always.  Try getting it from the token, then if it's not there (it should be unless there's misconfiguration), get it from the db.

Reply:
I've updated the scope of the ticket to include changing how we are currently implementing auth for this endpoint. I will make the changes requested.